web/src/hooks/mutations/useEditProjectMeta.ts

```diff
@@ -0,0 +1,38 @@
+import { useMutation } from '@tanstack/react-query';
+import { AxiosError } from 'axios';
+
+export const useEditProjectMeta = (
+  handleSubmit: () => Promise<any>,
+  resetForm: () => void,
+  toast: any,
+  queryClient: any,
+  namespace: string,
+  name: string,
+  tag: string,
+  onSuccessfulSubmit: () => void,
+  onFailedSubmit: () => void,
+  newTag: string,
+  newName: string
+) => {
+  return useMutation({
+    mutationFn: handleSubmit,
+    onSuccess: () => {
+      resetForm({}, { keepValues: true });
+      toast.success('Project metadata updated successfully.');
+      queryClient.invalidateQueries([namespace, name, tag]);
+      onSuccessfulSubmit();
+
+      if (newTag !== tag || newName !== name) {
+        window.location.href = `/${namespace}/${newName}?tag=${newTag}`;
+      }
+    },
+    onError: (error: AxiosError) => {
+      if (error.response?.status === 401) {
+        toast.error('You are not authorized to edit this project.');
```

Comment:
You might need to bring `toast` into scope here.

Nvm, I see you have it in the function params. Can we just import it here? Is that possible? that way we don't need to give this function `toast` each time we call it.